scripts/vendor_browserify.sh

```diff
@@ -1,22 +1,427 @@
 #!/usr/bin/env bash
 
+# Vendor in files from npm based on /third_party/npm/package.json & /third_party/npm/yarn.lock
+
 set -eu -o pipefail
 # -e: exits if a command fails
 # -u: errors if an variable is referenced before being set
 # -o pipefail: causes a pipeline to produce a failure return code if any command errors
 
-echo "Compiling browserify with ncc"
-./node_modules/.bin/ncc build node_modules/browserify/bin/cmd.js -o third_party/github.com/browserify/browserify
+# sedi makes `sed -i` work on both OSX & Linux
+# See https://stackoverflow.com/questions/2320564/i-need-my-sed-i-command-for-in-place-editing-to-work-with-both-gnu-sed-and-bsd
+sedi () {
+  case $(uname) in
+    Darwin*) sedi=('-i' '') ;;
+    *) sedi='-i' ;;
+  esac
+
+  sed "${sedi[@]}" "$@"
+}
+
+capture_and_run() {
+  CMDS="${CMDS}# "$(echo "$@" | sed "s#${RULES_NODEJS_DIR}/##g")$'\n'
+  "$@"
+}
+
+readonly RULES_NODEJS_DIR=$(cd $(dirname "$0")/..; pwd)
+readonly THIRD_PARTY_DIR=${RULES_NODEJS_DIR}/third_party
+readonly DATE=$(date)
+readonly PACKAGE_JSON_SHA256=$(cat ${THIRD_PARTY_DIR}/package.json | openssl dgst -sha256)
+readonly YARN_LOCK_SHA256=$(cat ${THIRD_PARTY_DIR}/yarn.lock | openssl dgst -sha256)
+
+readonly BROWSERIFY_PLUGINS=(
+  '@babel/plugin-transform-modules-commonjs'
+)
+
+(
+  cd ${THIRD_PARTY_DIR}
+  yarn install
+
+  ##############################################################################
+  # Vendor browserify
+  ##############################################################################
+  (
+    CMDS=""
+    readonly PKG="browserify"
+    readonly SRC_DIR=${THIRD_PARTY_DIR}/node_modules/${PKG}
+    readonly DST_DIR=${THIRD_PARTY_DIR}/npm/node_modules/${PKG}
+
+    echo "Vendoring ${PKG} to ${DST_DIR}"
+    rm -rf ${DST_DIR}
+    mkdir -p ${DST_DIR}
+
+    echo "- compiling ${PKG} with ncc"
+    capture_and_run ${THIRD_PARTY_DIR}/node_modules/.bin/ncc build ${SRC_DIR}/bin/cmd.js -o ${DST_DIR}
+
+    echo "- local mod: revert https://github.com/browserify/browserify/pull/1801"
+    capture_and_run sedi 's#parent.id !== self._mdeps.top.id#parent.id#' ${DST_DIR}/index.js
```

Comment:
in theory this won't work anymore - a PR is against the sources but now you mirror the NPM package, so if the package was authored in typescript we'd be applying a patch against the output JS instead.
other than that I don't have a good reason why we shouldn't mirror from npm rather than github...

Reply:
still works. I double checked. sources are `.js`. PR got merged but is not part of the latest release yet (`16.5.0`). released code is still
```
                if (opts.preserveSymlinks && parent.id !== self._mdeps.top.id) {
                    return cb(err, path.resolve(file), pkg, file)
                }
```